stellar/faucet.js

```diff
@@ -0,0 +1,55 @@
+const { Command, Option } = require('commander');
+const { ASSET_TYPE_NATIVE, getWallet, addBaseOptions, getBalances } = require('./utils');
+const { saveConfig, loadConfig, printInfo, printWarn, getChainConfig } = require('../common');
+const { Horizon } = require('@stellar/stellar-sdk');
+
+require('./cli-utils');
+
+async function processCommand(chain, options) {
+    const keyPair = await getWallet(chain, options);
+    const recipient = options.recipient || keyPair.publicKey();
+    const horizonServer = new Horizon.Server(chain.horizonRpc);
+    const balance = await getBalances(horizonServer, recipient).then((balances) =>
+        balances.find((balance) => balance.asset_type === ASSET_TYPE_NATIVE),
+    );
+
+    printInfo(`Requesting funds for`, recipient);
+
+    if (Number(balance?.balance || '0') >= Number(options.minBalance)) {
+        printWarn('Wallet balance above minimum, skipping faucet request');
+        process.exit(0);
+    }
+
+    await horizonServer.friendbot(recipient).call();
+
+    printInfo('Funds requested', recipient);
+}
+
+async function mainProcessor(options, processor) {
+    const config = loadConfig(options.env);
+    const chain = getChainConfig(config, options.chainName);
+    await processor(chain, options);
+    saveConfig(config, options.env);
```

Comment:
Does this script saves balances and sequence to config? if not, we should remove `saveConfig`.

Reply:
good point, removed this